lib/matplotlib/collections.py

```diff
@@ -899,6 +900,118 @@ def set_paths(self, paths):
     def get_paths(self):
         return self._paths
 
+    def legend_items(self, mode="colors", useall="auto", num=10,
+                     fmt=None, func=lambda x: x, **kwargs):
+        """
+        Creates legend handles and labels for a PathCollection. This is useful
+        for obtaining a legend for a :meth:`~.Axes.scatter` plot. E.g.::
+
+            scatter = plt.scatter([1,2,3], [4,5,6], c=[7,2,3])
+            plt.legend(*scatter.legend_items())
+
+        Also see the :ref:`automatedlegendcreation` example.
+
+        Parameters
+        ----------
+        mode : string, optional, default *"colors"*
+            Can be *"colors"* or *"sizes"*. In case of *"colors"*, the legend
+            handles will show the different colors of the collection. In case
+            of "sizes", the legend will show the different sizes.
+        useall : bool or string "auto", optional (default "auto")
+            If True, use all unique elements of the mappable array. If False,
+            target to use *num* elements in the normed range. If *"auto"*, try
+            to determine which option better suits the nature of the data.
+        num : int or None, optional (default 10)
+            Target number of elements to create in case of *useall=False*.
+            The number of created elements may slightly deviate from that due
+            to a `~.ticker.Locator` being used to find useful locations.
+        fmt : string, `~matplotlib.ticker.Formatter`, or None (default)
+            The format or formatter to use for the labels. If a string must be
+            a valid input for a `~.StrMethodFormatter`. If None (the default),
+            use a `~.ScalarFormatter`.
+        func : function, default *lambda x: x*
```

Comment:
I found this too confusing.  What does `*func*` do and why do you need it?  Maybe this needs an example...

Reply:
`func`  would allow to calculate the values to be shown in the legend. This is mainly useful for "sizes", because unlike the `c` parameter in scatter, which takes numbers and maps them to color using a `Normalize`, the `s` parameter does not have a normalization. This means that usually you would encode your data into sizes and provide those sizes in the scatter call. However in the legend you might still want to show the original data's values. Example: You use the  `s` to show price ranging between 0.5 and 10 dollars. Now you cannot use those numbers directly as scatter sizes, because that would result in tiny dots. So you convert them to sizes via `s = lambda price : 0.5*(price*10)^2`. Now `func` would allow you to provide `price = lambda s: np.sqrt(2*s)/10` to have a nice legend using the actual prices in dollars. 

I think you are right that an example might be useful here. I could add that in addition to the already extended example?

Reply:
Makes sense.  I was particularly flummoxed by "This converts the initial values for color or size and needs to take a numpy array as input."  I wasn't sure what "initial" meant.

For the doc string maybe:  "Function to calculate the labels.  Often the *size* (or *color*) argument to scatter will have been pre-processed by the user using a function ``s = f(x)`` to make the markers visible;  eg ``size = np.log10(x)`.  Providing this function allows that pre-processing to be inverted, so that the legend labels have the correct values.  eg ``np.exp(x, 10)``"

Reply:
I reformulated the docstring and also added an additional example now.